lib/cocoapods-core/source/aggregate.rb

```diff
@@ -198,26 +187,34 @@ def update_search_index(search_data)
       # @param  [Set] set
       #         The set for which the information is needed.
       #
-      # @note   If the specification can't load an empty hash is returned and
+      # @note   If the specification can't load an empty array is returned and
       #         a warning is printed.
       #
       # @note   For compatibility with non Ruby clients a strings are used
       #         instead of symbols for the keys.
       #
-      # @return [Hash{String=>String}] A hash with the search information.
+      # @return [Array<String>] An array of words contained by the set's search related information.
       #
-      def search_data_from_set(set)
-        result = {}
+      def word_list_from_set(set)
         spec = set.specification
-        result['version'] = spec.version.to_s
-        result['summary'] = spec.summary
-        result['description'] = spec.description
-        result['authors'] = spec.authors.keys.sort * ', '
-        result
+        word_list = [set.name.dup]
+        if spec.summary
+          word_list += spec.summary.split
```

Comment:
The `\w` was just an example – we can be as clever as we want to be when tokenizing, handling special cases like `Objective-C`. I'd like to leave it up to you as the main CLI search engine writer how clever exactly we want to be. A more conservative possibility would be to use `.scan(%r{[\w-]+})` with a liberal sprinkling of non-alphabetic characters added in addition to the `-`.